### DISCUSSION TITLE
archery可以使用普通用户操作数据库吗？

### DISCUSSION THREAD

**Discussion by a8159236** [hhyo/Archery#2279] [General]
archery可以使用普通用户操作数据库吗？
比如使用的是Mysql，创建了一个数据库A，然后创建了一个用户user1赋予数据库 A 的读写权限，就可以在archery中通过添加user1账户登录来对数据库 A 进行吗？除了读写权限还需要赋予其他的操作权限吗？

**Comment by a8159236** [comment]
我试过只给普通用户某个数据库的读写权限好像不行，要给超级权限，像这样：GRANT ALL PRIVILEGES ON *.* TO 'user'@'%'；，所以普通用户也要给整个数据库的supper权限吗？

**Comment by LeoQuote** [comment]
改表/上线 需要 `Replication Client`, `Replication Slave` 权限 https://github.com/hhyo/Archery/wiki/FAQ#%E6%89%A7%E8%A1%8C%E6%8F%90%E7%A4%BA%E6%97%A0%E6%B3%95%E8%8E%B7%E5%8F%96master-binlog%E4%BF%A1%E6%81%AF

查询的话只需要 select 的权限

**Reply by a8159236** [reply]
那生成回滚语句呢

**Reply by a8159236** [reply]
好像不给超级权限不自动生成回滚语句

**Reply by a8159236** [reply]
@LeoQuote

**Reply by LeoQuote** [reply]
我不太清楚了, 生成回滚语句是 goinception 的功能, 可以参考下 goinception 的手册

**Reply by a8159236** [reply]
好的谢谢
